pkg/snowflake/tag_association.go

```diff
@@ -76,12 +88,22 @@ func NewTagAssociationBuilder(tagID string) *TagAssociationBuilder {
 
 // Create returns the SQL query that will set the tag on an object.
 func (tb *TagAssociationBuilder) Create() string {
-	return fmt.Sprintf(`ALTER %v %v SET TAG "%v"."%v"."%v" = '%v'`, tb.objectType, tb.objectIdentifier, tb.databaseName, tb.schemaName, tb.tagName, EscapeString(tb.tagValue))
+	if strings.ToUpper(tb.objectType) == "COLUMN" {
+		tableName, columnName := tb.GetTableAndColumnName()
+		return fmt.Sprintf(`ALTER TABLE %v ALTER COLUMN %v SET TAG "%v"."%v"."%v" = '%v'`, tableName, columnName, tb.databaseName, tb.schemaName, tb.tagName, EscapeString(tb.tagValue))
```

Comment:
We should change the `ALTER COLUMN` with `MODIFY COLUMN`
```suggestion
		return fmt.Sprintf(`ALTER TABLE %v MODIFY COLUMN %v SET TAG "%v"."%v"."%v" = '%v'`, tableName, columnName, tb.databaseName, tb.schemaName, tb.tagName, EscapeString(tb.tagValue))
```